src/Common/src/CoreLib/System/IO/TextReader.cs

```diff
@@ -261,15 +265,29 @@ public virtual Task<int> ReadAsync(char[] buffer, int index, int count)
                     return t.Item1.Read(t.Item2.Span);
                 }, Tuple.Create(this, buffer), cancellationToken, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default));
 
+#if !MONO
```

Comment:
We want to use ValueTask, please undo this change

Reply:
The signature change is because StreamReader overrides this with internal `override async ValueTask<int> ReadAsyncInternal` which is not supported on MCS. Maybe we can make a MCS compilation define?

Reply:
We have it (for very long time), it's called `__MonoCS__`